pkg/packagekit/updates.jsx

```diff
@@ -444,11 +444,15 @@ class ApplyUpdates extends React.Component {
                     <strong>{ PK_STATUS_STRINGS[lastAction.status] || PK_STATUS_STRINGS[PK.Enum.STATUS_UPDATE] }</strong>
                     &nbsp;{lastAction.package}
                 </React.Fragment>);
-            logRows = this.state.actions.slice(0, -1).map(action => (
-                <tr key={action.package}>
-                    <th>{PK_STATUS_LOG_STRINGS[action.status] || PK_STATUS_LOG_STRINGS[PK.Enum.STATUS_UPDATE]}</th>
-                    <td>{action.package}</td>
-                </tr>));
+            logRows = [];
+            for (let i = 0; i < this.state.actions.length - 1; i++) {
```

Comment:
Doesn't this miss the last element? Did you mean `<=` or drop the `-1`? This is why I prefer `.map()`, no off-by-one errors :grinning: The second parameter of map() is the index, so you don't need a for loop for that.

Reply:
> Doesn't this miss the last element?

Originally it was `this.state.actions.slice(0, -1).map()` which (if I read it correctly) goes from first to second to last elements, right? (The last element is treated specially).

>The second parameter of map() is the index

ha, nice! good to know :)

Reply:
You are right, this is supposed to exclude the last element. `lastAction` is already being rendered separately above.

Reply:
Switched to using map